scripts/end_to_end_test/run_end_to_end_test.py

```diff
@@ -413,6 +433,133 @@ def add_node(parameters, test_instance):
+        api.sync(api.tokens.wealth(entity, amount))
+
+
+def create_contract(parameters, test_instance):
+    from smart_contract_tests.synergetic_utils import ContractTestHelper
+    nodes = parameters["nodes"]
+    name = parameters["name"]
+    fee_limit = parameters["fee_limit"]
+    for node_index in nodes:
+        node_host = "localhost"
+        node_port = test_instance._nodes[node_index]._port_start
+
+        api = LedgerApi(node_host, node_port)
+
+        # create the entity from the node's private key
+        entity = Entity(get_nodes_private_key(test_instance, node_index))
+
+        helper = ContractTestHelper(
+            name, api, entity, test_instance._workspace)
+        helper.create_new(fee_limit)
+        test_instance._nodes[node_index]._contract = helper
+
+
+def run_contract(parameters, test_instance):
+    from smart_contract_tests.synergetic_utils import ContractTestHelper
+    nodes = parameters["nodes"]
+    contract_name = parameters["contract_name"]
+    wait_for_blocks_num = parameters["wait_for_blocks"]
+    for node_index in nodes:
+        node_host = "localhost"
+        node_port = test_instance._nodes[node_index]._port_start
+
+        api = LedgerApi(node_host, node_port)
+
+        # create the entity from the node's private key
+        entity = Entity(get_nodes_private_key(test_instance, node_index))
+
+        try:
+            contract_helper = test_instance._nodes[node_index]._contract
+        except AttributeError:
+            output(
+                f"No contract stored in test_instance (node_index={node_index})! Loading from file...")
+            contract_helper = ContractTestHelper(
+                contract_name, api, entity, test_instance._workspace)
+            contract_helper.load()
+
+        contract_helper.submit_random_data(10, (0, 200))
+        api.wait_for_blocks(wait_for_blocks_num)
+        valid = contract_helper.validate_execution()
+        if not valid:
+            output(
+                f"Synergetic contract ({contract_name}) execution failed on node {node_index}!")
+            raise Exception(
+                f"Synergetic contract ({contract_name}) execution failed on node {node_index}!")
+        else:
+            output(
+                f"Synergetic contract ({contract_name}) executed on node {node_index} ")
+
+
+def wait_for_blocks(parameters, test_instance):
+    nodes = parameters["nodes"]
+    wait_for_blocks_num = parameters["num"]
+    for node_index in nodes:
+        node_host = "localhost"
+        node_port = test_instance._nodes[node_index]._port_start
+        output(
+            f"Waiting for {wait_for_blocks_num} blocks on node {node_index}")
+        api = LedgerApi(node_host, node_port)
+        api.wait_for_blocks(wait_for_blocks_num)
+
+
+def verify_chain_sync(parameters, test_instance):
+    from fetch.cluster.chain import ChainSyncTesting
+    max_trials = parameters["max_trials"]
+    sleep_time = parameters["sleep_between_trials"]
```

Comment:
I wouldn't pass these in, rather it can be calculated based on the block time. It is fine to take a long time I think since the test has a timeout.